src/main.ts

```diff
@@ -231,34 +235,130 @@ const makeContainerPublic = async (session: Session, containerId: string): Promi
     )
 }
 
-export const resolveLatestLockboxVersions = async (session: Session, secrets: Secret[]): Promise<Secret[]> => {
-    const secretsWithLatest = secrets.filter(s => s.versionId === 'latest')
-    if (secretsWithLatest.length === 0) {
-        return secrets
-    }
+type ResolutionResult =
+    | { status: 'success'; secret: Secret }
+    | { status: 'fallback'; original: Secret }
+    | { status: 'error'; error: Error }
+
+const resolveSecretsById = async (session: Session, secrets: Secret[]): Promise<ResolutionResult[]> => {
     const client = session.client(secretService.SecretServiceClient)
-    const { results: resolvedSecrets, errors: resolveErrors } = await PromisePool.for(secretsWithLatest)
+    const { results } = await PromisePool.for(secrets)
         .withConcurrency(5)
-        .process(async secret => {
-            const lockboxSecret: LockboxSecret = await client.get({ secretId: secret.id })
+        .process(async (secret): Promise<ResolutionResult> => {
+            let lockboxSecret: LockboxSecret
+            try {
+                lockboxSecret = await client.get({ secretId: secret.id })
+            } catch {
+                return { status: 'fallback', original: secret }
+            }
+
             if (!lockboxSecret.currentVersion) {
-                throw new Error(`Secret ${secret.id} has no current version`)
+                return {
+                    status: 'error',
+                    error: new Error(`Secret ${secret.id} has no current version`)
+                }
             }
             return {
-                ...secret,
-                versionId: lockboxSecret.currentVersion.id
+                status: 'success',
+                secret: {
+                    ...secret,
+                    versionId: lockboxSecret.currentVersion.id
+                }
             }
         })
-    if (resolveErrors.length > 0) {
-        const errorMessages = resolveErrors.map(e => e.message).join(', ')
+    return results
+}
+
+const findSecretsInFolder = async (session: Session, folderId: string): Promise<Map<string, LockboxSecret>> => {
+    const client = session.client(secretService.SecretServiceClient)
+    const folderSecretsMap = new Map<string, LockboxSecret>()
+    let pageToken: string | undefined = undefined
+
+    do {
+        const resp: ListSecretsResponse = await client.list(
+            ListSecretsRequest.fromPartial({
+                folderId,
+                pageSize: 100,
+                pageToken: pageToken || ''
+            })
+        )
+        if (resp.secrets) {
+            for (const secret of resp.secrets) {
+                folderSecretsMap.set(secret.name, secret)
+            }
+        }
+        pageToken = resp.nextPageToken
+    } while (pageToken)
+
+    return folderSecretsMap
+}
+
+export const resolveLatestLockboxVersions = async (
```

Comment:
`resolveLatestLockboxVersions` now assumes `PromisePool` preserves input order (`results[index]` matched to `secretsWithLatest[index]` and fallback indices). PromisePool collects results in completion order unless `.useCorrespondingResults()` is set, so secrets can be paired with the wrong result or error (e.g., first-fastest secret resolves the slot for another). This can return incorrect secret IDs/versionIds or throw on the wrong secret.